### PR TITLE
docs: Add new spec

### PR DESCRIPTION
**Pull request by stephen304**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    RequireApproval: false
+    StateDir: "/var/lib/autoygg"
+    MaxClients: 10
+    LeaseTimeoutSeconds: 14400
+    GatewayTunnelIP: "10.42.0.1"
+    GatewayTunnelNetmask: 16
+    GatewayTunnelIPRangeMin: "10.42.42.1"
+    GatewayTunnelIPRangeMax: "10.42.42.255"
+    ACLEnabled: true
+    ACLMode: "blacklist"/"whitelist"
```
Comment (cure) [inline SPEC.md]: How about we do this a little differently. What if we just had an acl struct, something like this:

type acl struct {
  YggIP string
  action string // Whitelisted or Blacklisted
  comment string
}

The question that remains, then, is what 'whitelisted' means. Does it mean they get to bypass registration? That they are pre-approved but still need registration? Should it depend on the mode we're running in? What do you think?

If we do something along those lines, we don't need ACL entries in the config file, right?

Comment (stephen304) [reply SPEC.md]: We could do something like that and I guess that would help simplify the live reloading aspect of the acl files currently. If we keep calling it blacklist / whitelist then I think the most intuitive thing to do is allow registration/lease for the whitelist action, to keep it mirroring the blacklist behavior (being on or off the blacklist doesn't change the registration requirement, just controls access to the registration/lease endpoints). We'll also want a whitelist toggle, and the blacklist can just be always on.

Comment (cure) [reply SPEC.md]: I was thinking more a long the lines of one acl file. I would like to do away with white/blacklist separation. Each entry can specify 'block' (aka blacklist), 'approval', 'registration' or 'lease'. 

And if we keep thinking along those lines, why don't we allow for '*' in the YggIP field, then we could express the various modes that way, too (* / approval means everyone needs to register and be approved, * / registration means anyone can register (no approval required); * / lease is anonymous mode, * / block means nothing is allowed). We just have to make clear how the file is parsed (as in, do we read from top to bottom and enforce the first or last rule that applies to an IP, or do we use the 'least privilege' rule if multiple rules apply, or do we apply the most specific rule (i.e. an explicit IP vs. the wildcard catchall).

What do you think?

**Comment by cure** [comment]
OK, merging this now, we'll iterate.

**Pull request merged**